cli/azd/pkg/infra/provisioning/bicep/prompt.go

```diff
@@ -222,12 +238,25 @@ func (p *BicepProvider) promptForParameter(
 	help, _ := param.Description()
 	azdMetadata, _ := param.AzdMetadata()
 	paramType := p.mapBicepTypeToInterfaceType(param.Type)
+	// parameter `location` is a special case for azd, as it can be set in the environment
+	// and its value must be persisted in the .env file as AZURE_LOCATION
+	isLocationParam := key == "location"
```

Comment:
Supposedly, if `location` isn't mapped to `AZURE_LOCATION`, would the `location` variable name be the way we'd go about it?

Just wondering out loud since this may not be immediately possible today and we'd scope it out, but seems more likely to lead to a path of success.

Reply:
I think we can make this better by finding the parameter mapped to `AZURE_LOCATION`
We can do that as a follow up PR

In that world, AZD would handle parameters prompt as locations when they are decorated by location-metadata or when they are mapped to AZURE_LOCATION, regardless of the name.

I don't think there is any case right now, out there, from someone using `location` as parameter w/o mapping it to AZURE_LOCATION

Reply:
https://github.com/Azure/azure-dev/issues/5031